apps/backoffice-v2/src/common/components/molecules/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,22 @@
+import { DropdownMenu } from '@/common/components/molecules/DropdownMenu/DropdownMenu';
+import { DropdownMenuTrigger } from '@/common/components/molecules/DropdownMenu/DropdownMenu.Trigger';
+import { DropdownMenuContent } from '@/common/components/molecules/DropdownMenu/DropdownMenu.Content';
+import { isNonEmptyArray } from '@ballerine/common';
+import { DropdownMenuItem } from '@/common/components/molecules/DropdownMenu/DropdownMenu.Item';
+import React from 'react';
+import { DropdownProps } from '@/common/components/molecules/Dropdown/interfaces';
+
+export const Dropdown = <TItem,>({ options, trigger, props, children }: DropdownProps<TItem>) => {
+  return (
+    <DropdownMenu {...props?.menu}>
+      <DropdownMenuTrigger {...props?.trigger}>{trigger}</DropdownMenuTrigger>
+      <DropdownMenuContent {...props?.content}>
+        <>
+          {isNonEmptyArray(options) &&
+            options.map(item => children?.({ item, DropdownItem: DropdownMenuItem }))}
+          {!isNonEmptyArray(options) && <DropdownMenuItem>No options</DropdownMenuItem>}
```

Comment:
Same check performed two times, ternary operator won't fit here?

Reply:
I find ternaries in JSX ruin formatting and makes it hard to read and follow. I am considering `ts-pattern` though. 🤤